hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/state/ReadableState.java

```diff
@@ -15,43 +15,52 @@
  */
 package com.hedera.node.app.spi.state;
 
-import java.time.Instant;
 import java.util.Optional;
+import javax.annotation.Nonnull;
 
 /**
- * Provides access to key/value state for a service implementation.
+ * Provides access to key/value state for a service implementation. This interface is implemented by
+ * the Hedera application, and provided to the service implementation at the appropriate times. The
+ * methods of this class provide read-only access to the state.
  *
- * @param <K> The key
- * @param <V> The value
+ * @param <K> The key, which must be of the appropriate kind depending on whether it is stored in
+ *     memory, or on disk.
+ * @param <V> The value, which must be of the appropriate kind depending on whether it is stored in
+ *     memory, or on disk.
  */
-public interface State<K, V> {
+public interface ReadableState<K, V> {
```

Comment:
In general I do not get why this is part of an SPI package. From my point of view this is an API

Reply:
I'm not sure the distinction is useful. Every class, interface, etc. in these SPI packages exist so that services can be implemented. You cannot implement a service without ReadableState. Yes, some interfaces are meant to be implemented by the app and some by services, and we could segregate which are which, but it seems like overkill to have another module just for that. We could put them in different packages, but what would the full package names be?